client.go

```diff
@@ -371,6 +420,29 @@ func (co *Conn) WriteMsg(m *Msg) (err error) {
 // Write implements the net.Conn Write method.
 func (co *Conn) Write(p []byte) (n int, err error) {
 	switch t := co.Conn.(type) {
+	case *dohConn:
+		req, err := http.NewRequest(http.MethodPost, t.address, bytes.NewReader(p))
+		if err != nil {
+			return 0, err
+		}
+
+		req.Header.Set("Content-Type", dohMimeType)
+		req.Header.Set("Accept", dohMimeType)
+
+		d := t.client.dialTimeout() + t.client.writeTimeout() + t.client.readTimeout()
+
+		ctx, cancel := context.WithTimeout(context.Background(), d)
+		defer cancel()
+
+		req = req.WithContext(ctx)
+
+		resp, err := http.DefaultClient.Do(req)
```

Comment:
It would be useful to allow the `http.Client` to be specified, perhaps by adding a field to `Conn`.